hack/make-rules/test-cmd-util.sh

```diff
@@ -1613,7 +1613,7 @@ run_secrets_test() {
 
   ### Create a docker-registry secret in a specific namespace
   # Pre-condition: no SECRET exists
-  kube::test::get_object_assert 'secrets --namespace=test-secrets' "{{range.items}}{{$id_field}}:{{end}}" ''
+  kube::test::wait_object_assert 'secrets --namespace=test-secrets' "{{range.items}}{{$id_field}}:{{end}}" ''
```

Comment:
has there been discussion about making a single test-cmd suite be used against both federation and regular clusters? I feel like this is the first of many places where the tests will need to accomodate behavior differences between the two servers. In this case, I'd want to know if the core apiserver suddenly started retaining secrets after deletion, but this retry would mask that.